addons/shape-opacity/userscript.js

```diff
@@ -0,0 +1,61 @@
+export default async function ({ addon, global, console, msg }) {
+  let appended = false;
+  const clss = addon.tab.scratchClass;
+
+  const labelContainer = document.createElement("label");
+  labelContainer.className = clss("label_input-group");
+
+  const opacityText = document.createElement("span");
+  opacityText.className = clss("label_input-label");
+  opacityText.innerText = msg("opacity");
+
+  const opacityInput = document.createElement("input");
+  opacityInput.min = 0;
+  opacityInput.max = 100;
+  opacityInput.type = "number";
+
+  labelContainer.append(opacityText, opacityInput);
+
+  addon.tab.redux.initialize();
+  addon.tab.redux.addEventListener("statechanged", ({ detail }) => {
+    const { action } = detail;
+    if (action.type === "scratch-paint/select/CHANGE_SELECTED_ITEMS") {
+      if (action.selectedItems.length === 0) {
+        labelContainer.style.display = "none";
+      } else {
+        // Scratch returns undefined for bitmapMode when in bitmap mode...
+        if (action.bitmapMode !== false) return;
+        labelContainer.style.display = "";
+        if (!appended) {
+          appended = true;
+          const appendRow = document.querySelector("[class*=paint-editor_mod-mode-tools]");
+          appendRow.append(labelContainer);
```

Comment:
Could the input be next to Fill and Outline? That's where it makes the most sense. For consistency it should be present but disabled when no object is selected.